app/code/core/Mage/Sales/Model/Api2/Order/Comment/Rest/Admin/V1.php

```diff
@@ -27,4 +27,67 @@
  */
 class Mage_Sales_Model_Api2_Order_Comment_Rest_Admin_V1 extends Mage_Sales_Model_Api2_Order_Comment_Rest
 {
+    /**
+     * Add comment to order
+     *
+     * @param array $data
+     * @return string
+     */
+    protected function _create(array $data)
+    {
+        $orderId = $this->getRequest()->getParam(self::PARAM_ORDER_ID);
+        $order   = $this->_loadOrderById($orderId);
+
+        $status         = $data['status'] ?? false;
+        $comment        = $data['comment'] ?? '';
+        $visibleOnFront = $data['is_visible_on_front'] ?? 0;
+        $notifyCustomer = array_key_exists('is_customer_notified', $data) ? $data['is_customer_notified'] : false;
+
+        $historyItem = $order->addStatusHistoryComment($comment, $status);
+        $historyItem->setIsCustomerNotified($notifyCustomer)
+            ->setIsVisibleOnFront((int) $visibleOnFront)
+            ->save();
+
+        try {
+            if ($notifyCustomer && $comment) {
+                $oldStore = Mage::getDesign()->getStore();
+                $oldArea = Mage::getDesign()->getArea();
+                Mage::getDesign()->setStore($order->getStoreId());
+                Mage::getDesign()->setArea('frontend');
+            }
+
+            $order->save();
+            $order->sendOrderUpdateEmail((bool) $notifyCustomer, $comment);
+
+            if ($notifyCustomer && $comment) {
+                Mage::getDesign()->setStore($oldStore);
+                Mage::getDesign()->setArea($oldArea);
+            }
+        } catch (Mage_Core_Exception $e) {
+            $this->_critical($e->getMessage(), self::RESOURCE_INTERNAL_ERROR);
+        } catch (Throwable $t) {
+            Mage::logException($t);
+            $this->_critical($t->getMessage(), self::RESOURCE_UNKNOWN_ERROR);
+        }
+
+        return $this->_getLocation($historyItem);
```

Comment:
Also I later discovered the original PR had an error at this line after you add a comment. Apparently no one tested it.

Reply:
Mhh, i'd still merge it to have less differences to v20.0.